### DISCUSSION TITLE
Can we retain the hash value while navigating

### DISCUSSION THREAD

**Discussion by mrhammadasif** [vuejs/router#1724] [Help and Questions] [answered]
While navigating `router.push({name: 'somepage'})` to other pages in the Application, I want to persist the `#userId=223` hash value.

For example, If I want to navigate from "http://localhost:8080/users#userId=10" to "http://localhost:8080/profile#userId=20", How can I retain the value of hash

**Comment by posva** [comment, marked as answer]
You can:

```ts
router.push({ name: 'somepage', hash: route.hash })
```

**Reply by mrhammadasif** [reply]
That is understood but that would need to update all of my `router.push` statements. I am just thinking of a way to do it (like in middleware or something without a refresh in the browser)

Currently, I am doing it in the following way:

- in `onRouteEnter` global middleware, check if the hash value is present
- if not, navigate the current route with the new hash value

But the problem is the double loading of the same page. It caused a refresh of the complete webpage

**Reply by posva** [reply]
if you want to override all `router.push()`, create your own push method and override the `router`'s.

```ts
const originalPush = router.push
router.push = function (...args) {
  return original.push.call(this, ....)
}
```

**Reply by mrhammadasif** [reply]
Why didn't that come up in my mind, feeling so dumb right now :-D

Thank you so much
